tools/pipeline_perf_test/load_generator/loadgen.py

```diff
@@ -303,6 +304,72 @@ def syslog_worker_thread(self, thread_id: int, args: dict) -> None:
 
         sock.close()
 
+    def syslog_udp_worker_thread(self, thread_id: int, args: dict) -> None:
+        """
+        Worker thread that sends syslog messages to a syslog server via UDP.
+        """
+        syslog_server = os.getenv("SYSLOG_SERVER", "localhost")
+        syslog_port = int(os.getenv("SYSLOG_PORT", "514"))
+
+        print(f"Thread {thread_id}: Using UDP transport to syslog server {syslog_server}:{syslog_port}")
+
+        # Create UDP socket for syslog
+        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+
+        batch_size = args["batch_size"]
+        thread_count = args["threads"]
+        target_rate = args.get("target_rate")
+
+        if target_rate:
+            thread_rate = target_rate / thread_count
+            batch_interval = batch_size / thread_rate
+            print(
+                f"Thread {thread_id} started with rate limit: {thread_rate} "
+                f"logs/sec (interval: {batch_interval:.4f}s)"
+            )
+        else:
+            batch_interval = None
+            print(f"Thread {thread_id} started with no rate limit")
+
+        # Pre-generate syslog messages batch
+        syslog_batch = []
+        for _ in range(batch_size):
+            syslog_message = self.create_syslog_message(
+                body_size=args["body_size"],
+            )
+            syslog_batch.append(syslog_message)
+
+        batch_total_size = sum(len(msg) for msg in syslog_batch)
+
+        next_send_time = time.perf_counter()
+        while not self.stop_event.is_set():
+            try:
+                # UDP: Send individual messages instead of single batch of
+                # messages.
+                for message in syslog_batch:
+                    print(f"Sending syslog message {message} via UDP")
```

Comment:
Could we remove the `print` statement here? It would be adding contention among threads.

Reply:
leftover from my test. removing.